RecoTauTag/RecoTau/plugins/PFRecoTauDiscriminationAgainstMuonSimple.cc

```diff
@@ -0,0 +1,311 @@
+  bool dRmuonMatchLimitedToJetArea_;
+  double minPtMatchedMuon_;
+  int maxNumberOfMatches_;
+  std::vector<int> maskMatchesDT_;
+  std::vector<int> maskMatchesCSC_;
+  std::vector<int> maskMatchesRPC_;
+  int maxNumberOfHitsLast2Stations_;
+  std::vector<int> maskHitsDT_;
+  std::vector<int> maskHitsCSC_;
+  std::vector<int> maskHitsRPC_;
+  int maxNumberOfSTAMuons_, maxNumberOfRPCMuons_;
+
+  mutable int numWarnings_;
+  int maxWarnings_;
+  int verbosity_;
+};
+
+void PFRecoTauDiscriminationAgainstMuonSimple::beginEvent(const edm::Event& evt, const edm::EventSetup& es) 
+{
+  evt.getByToken(Muons_token, muons_);
+}
+
+namespace
+{
+  /* MB: not used in current implementation, but keep it if needed in future
+  const reco::Track* getTrack(const reco::Candidate& cand) {
+    const pat::PackedCandidate* pCand = dynamic_cast<const pat::PackedCandidate*>(&cand);
+    if ( pCand != nullptr && pCand->hasTrackDetails() )
+      return &pCand->pseudoTrack();
+    return nullptr;
+  }
+  */
+  void countHits(const pat::Muon& muon, std::vector<int>& numHitsDT, std::vector<int>& numHitsCSC, std::vector<int>& numHitsRPC)
+  {
+    if ( muon.outerTrack().isNonnull() ) {
+      const reco::HitPattern &muonHitPattern = muon.outerTrack()->hitPattern();
+      for (int iHit = 0; iHit < muonHitPattern.numberOfAllHits(reco::HitPattern::TRACK_HITS); ++iHit) {
+          uint32_t hit = muonHitPattern.getHitPattern(reco::HitPattern::TRACK_HITS, iHit);
+	if ( hit == 0 ) break;	    
+	if ( muonHitPattern.muonHitFilter(hit) && (muonHitPattern.getHitType(hit) == TrackingRecHit::valid || muonHitPattern.getHitType(hit) == TrackingRecHit::bad) ) {
+	  int muonStation = muonHitPattern.getMuonStation(hit) - 1; // CV: map into range 0..3
+	  if ( muonStation >= 0 && muonStation < 4 ) {
+	    if      ( muonHitPattern.muonDTHitFilter(hit)  ) ++numHitsDT[muonStation];
+	    else if ( muonHitPattern.muonCSCHitFilter(hit) ) ++numHitsCSC[muonStation];
+	    else if ( muonHitPattern.muonRPCHitFilter(hit) ) ++numHitsRPC[muonStation];
+	  }
+	}
+      }
+    }
+  }
+
+  void countMatches(const pat::Muon& muon, std::vector<int>& numMatchesDT, std::vector<int>& numMatchesCSC, std::vector<int>& numMatchesRPC)
+  {
+    const std::vector<reco::MuonChamberMatch>& muonSegments = muon.matches();
+    for ( std::vector<reco::MuonChamberMatch>::const_iterator muonSegment = muonSegments.begin();
+	  muonSegment != muonSegments.end(); ++muonSegment ) {
+      if ( muonSegment->segmentMatches.empty() ) continue;
+      int muonDetector = muonSegment->detector();
+      int muonStation = muonSegment->station() - 1;
+      assert(muonStation >= 0 && muonStation <= 3);
+      if      ( muonDetector == MuonSubdetId::DT  ) ++numMatchesDT[muonStation];
+      else if ( muonDetector == MuonSubdetId::CSC ) ++numMatchesCSC[muonStation];
+      else if ( muonDetector == MuonSubdetId::RPC ) ++numMatchesRPC[muonStation];      
+    }
+  }
+
+  std::string format_vint(const std::vector<int>& vi)
```

Comment:
this method and also countMatches and countHits are the same as in the \*Muon2.cc.
They can be simply moved to a common implementation place and reused